packages/react/src/ElementsInput/ElementsInput.tsx

```diff
@@ -43,9 +43,26 @@ export function ElementsInput(props: ElementsInputProps): JSX.Element {
       props.onChange(newValue);
     }
   }
+
+  const orderedElements = useMemo<[string, InternalSchemaElement][]>(() => {
+    const result = Object.entries(elements);
+    result.sort((a, b) => {
+      const aElem = a[1];
+      const bElem = b[1];
+
+      // required elements go first
+      if (aElem.min > 0 || bElem.min > 0) {
```

Comment:
We should discuss w/ @rahul1.  I definitely see the logic behind this, but it will a be a functional change for some users who use `app` daily (how many times they need to tab, etc).  It *should* be minor, but we should get a sanity check.

Reply:
Ah, yeah, that's a good call. This is probably a more significant breaking change. I'll make an opt-in prop for now.

Reply:
Err, scratch that; making it optional would require passing the prop down several layers (or introducing a new `ResourceFormContext` which doesn't seem worth it at this point. Gonna remove this and keep it on the back burner for now.